x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -305,22 +315,33 @@ func (m *MetricSet) readCloudwatchConfig() (listMetricWithDetail, map[string][]n
 
 			if config.ResourceTypeFilter != "" {
 				if _, ok := resourceTypesWithTags[config.ResourceTypeFilter]; ok {
-					resourceTypesWithTags[config.ResourceTypeFilter] = config.Tags
-
+					if config.TagsFilter != nil {
```

Comment:
WDYT about doing this logic once at the beginning of this function and then use a variable for tags in all the subsequent uses?

Reply:
Good point!